### PR TITLE
Dreamview: downsample prediction points

### PR DESCRIPTION
**Pull request by siyangy**

### CODE REVIEW

File: modules/dreamview/backend/simulation_world/simulation_world_service.cc [hunk 1]
```diff
@@ -813,10 +812,18 @@ void SimulationWorldService::CreatePredictionTrajectory(
   for (const auto &traj : obstacle.trajectory()) {
     Prediction *prediction = world_object->add_prediction();
```
Comment (lianglia-apollo) [inline modules/dreamview/backend/simulation_world/simulation_world_service.cc]: The function CreatePredictionTrajectory has a wrong order for input/output variables.

File: modules/dreamview/backend/simulation_world/simulation_world_service.cc [hunk 2]
```diff
@@ -853,10 +860,8 @@ void SimulationWorldService::UpdateSimulationWorld(
...
     // Downsample the path points for frontend display.
-    // Angle threshold is about 5.72 degree.
-    constexpr double angle_threshold = 0.1;
-    std::vector<int> sampled_indices =
-        DownsampleByAngle(path.path_points(), angle_threshold);
+    auto sampled_indices =
+        DownsampleByAngle(path.path_points(), kAngleThreshold);
 
     route_paths_.emplace_back();
     RoutePath *route_path = &route_paths_.back();
```
Comment (lianglia-apollo) [inline modules/dreamview/backend/simulation_world/simulation_world_service.cc]: route_paths_ is a vector, the pointer address can change.

File: modules/dreamview/backend/simulation_world/simulation_world_service.cc (hunk 1 above)
Comment (siyangy) [reply modules/dreamview/backend/simulation_world/simulation_world_service.cc]: done

File: modules/dreamview/backend/simulation_world/simulation_world_service.cc (hunk 2 above)
Comment (siyangy) [reply modules/dreamview/backend/simulation_world/simulation_world_service.cc]: not sure I get this?

Comment (startcode) [reply modules/dreamview/backend/simulation_world/simulation_world_service.cc]: there will be a problem only when receiving two routing_responese simultaneously, which is unlikely for routing topic.

**Pull request merged**